nirum_wsgi.py

```diff
@@ -8,21 +8,45 @@
 import re
 import typing
 
+from nirum.datastructures import List
 from nirum.deserialize import deserialize_meta
 from nirum.exc import (NirumProcedureArgumentRequiredError,
                        NirumProcedureArgumentValueError)
 from nirum.serialize import serialize_meta
 from nirum.service import Service
 from six import integer_types
 from six.moves import reduce
-from werkzeug.exceptions import HTTPException
 from werkzeug.http import HTTP_STATUS_CODES
-from werkzeug.routing import Map, Rule
 from werkzeug.serving import run_simple
 from werkzeug.wrappers import Request, Response
 
 __version__ = '0.2.0'
-__all__ = 'WsgiApp',
+__all__ = 'AnnotationError', 'NoJsonError', 'ServiceMethodError', 'WsgiApp'
+
+
+def parse_json_payload(request):
+    payload = request.get_data(as_text=True)
+    if payload:
+        try:
+            json_payload = json.loads(payload)
+        except (TypeError, ValueError):
+            raise NoJsonError(payload)
+        else:
+            return json_payload
+    else:
+        return {}
+
+
+class NoJsonError(ValueError):
```

Comment:
Seems `InvalidJsonError` would be more suitable for the use of this exception type.